### PR TITLE
 Add exceptions stack to VirtualMachine

### LINKED ISSUE #867: Continue in except clause broken
The following, raises an exception `TypeError: Unsupported method: __next__`:
```python
for _ in [1, 2]:
    try:
        raise ArithmeticError()
    except ArithmeticError as e:
        continue
```
The cause of this is that the exception is put on the top of the stack, but continue expects the iterator to be on the top of the stack.

### PR DESCRIPTION
**Pull request by palaviv**
This resolves #867 and resolves [#837](https://github.com/RustPython/RustPython/pull/831#discussion_r275260542)

### CODE REVIEW

File: vm/src/frame.rs [hunk 1]
```diff
@@ -860,6 +860,10 @@ impl Frame {
                 self.push_value(formatted);
                 Ok(None)
             }
+            bytecode::Instruction::PopException {} => {
+                assert!(!vm.pop_exception().is_none());
```
Comment (coolreader18) [inline vm/src/frame.rs]: This can just be `assert!(vm.pop_exception().is_some())`.

File: vm/src/compile.rs [hunk 2]
```diff
@@ -330,24 +328,15 @@ impl Compiler {
                     match cause {
                         Some(cause) => {
                             self.compile_expression(cause)?;
-                            self.emit(Instruction::Raise {
-                                argc: 2,
-                                in_exc: self.in_exc_handler,
-                            });
+                            self.emit(Instruction::Raise { argc: 2 });
```
Comment (windelbouwman) [inline vm/src/compile.rs]: This looks cleaner already!

File: vm/src/compile.rs [hunk 3]
```diff
@@ -597,8 +583,6 @@ impl Compiler {
         self.emit(Instruction::Jump { target: else_label });
 
         // except handlers:
-        let was_in_exc_handler = self.in_exc_handler;
-        self.in_exc_handler = true;
```
Comment (windelbouwman) [inline vm/src/compile.rs]: Its really good to lose this state boolean!

File: vm/src/frame.rs [hunk 4]
```diff
@@ -966,7 +970,8 @@ impl Frame {
         while let Some(block) = self.pop_block() {
             match block.typ {
                 BlockType::TryExcept { handler } => {
-                    self.push_value(exc);
+                    self.push_value(exc.clone());
```
Comment (windelbouwman) [inline vm/src/frame.rs]: Its a bit weird to push the exception on the value stack as well as on the exception stack, but I guess we require this for proper value stack unwinding.

File: vm/src/frame.rs (hunk 1 above)
Comment (palaviv) [reply vm/src/frame.rs]: Done

**Pull request merged**